### PR TITLE
Introduce registerModel(), remove schwifty(), no models during registration

### PR DESCRIPTION
**Pull request by devinivy**
Implement changes described in #91.

### CODE REVIEW

File: API.md [hunk 1]
```diff
@@ -6,26 +6,21 @@
 
 ## The hapi plugin
 ### Registration
-Schwifty may be registered multiple times—it should be registered in any plugin that would like to use any of its features.  Upon each registration these options are collected until server initialization.  Models and knex configurations passed during configuration are attributed to the registering plugin/server under schwifty's [ownership system](#plugin-ownership-of-knex-instances-and-models). The model(s) `name` must be unique across the entire server.  Server initialization will fail if any knex instances handled by schwifty do not have basic database connectivity.
+Schwifty may be registered multiple times—it should be registered in any plugin that would like to use any of its features.  Upon each registration these options are collected until server initialization.  Knex configurations passed during configuration are attributed to the registering plugin/server under schwifty's [ownership system](#plugin-ownership-of-knex-instances-and-models).  Server initialization will fail if any knex instances handled by schwifty do not have basic database connectivity.
```
Comment (devinivy) [inline API.md:9]:
```suggestion
Schwifty may be registered multiple times—it should be registered in any plugin that would like to use any of its features.  Upon each registration these options are collected until server initialization.  Knex configurations passed during registration are attributed to the registering plugin/server under schwifty's [ownership system](#plugin-ownership-of-knex-instances-and-models).  Server initialization will fail if any knex instances handled by schwifty do not have basic database connectivity.
```

**Review by YoannMa — APPROVED** [review]

File: lib/schema.js [hunk 2]
```diff
@@ -2,22 +2,14 @@
...
     teardownOnStop: Joi.boolean(),
     migrateOnStart: Joi.boolean().truthy('latest').allow('rollback')
 });
 
-exports.schwifty = Joi.alternatives([
-    Joi.array().items(internals.model).single(),
-    internals.configBase
-]);
+exports.registerModel = Joi.array()
+    .items(Joi.func().class())
```
Comment (wswoodruff) [inline lib/schema.js:13]: This line's got `class`! 🧐

**Comment by devinivy** [comment]
I appreciate the review, pals!

File: API.md (hunk 1 above)
Comment (devinivy) [reply API.md:9]: I forgot to apply this 🤦 I wont lose this change though, pushing it up shortly.

**Pull request merged**